lib/steps.js

```diff
@@ -419,30 +443,44 @@ Then('I arrive at the next page', async () => {  // √
   /* Tests for detection of url change from button or link tap.
   *    Can other things trigger navigation? Re: other inputs things
   *    like 'Enter' acting like a click is a test for docassemble */
-  expect( scope.navigated ).to.be.true;
+  let msg = `User did not arrive at the next page.`
+  if ( !scope.navigated ) { await scope.addToReport(scope, { type: `error`, value: msg }); }
+  expect( scope.navigated, msg ).to.be.true;
   await scope.afterStep(scope);
 });
 
-Then('I should see the link to {string}', async (url) => {  // √
-  let link = await scope.page.$(`*[href="${ url }"]`);
-  expect(link, `Missing a link to ${ url }`).to.exist;
-  
+// Link observations have the 'Escape' button link in mind
+Then('I should see the link {string}', async ( linkText ) => {  // √
+  /* Loosley match link visible text. Should probably deprecate this. */
+  let [link] = await scope.page.$x(`//a[contains(text(), "${linkText}")]`);
+
+  let msg = `Cannot find a link with the text "${ linkText }".`;
+  if ( !link ) { await scope.addToReport(scope, { type: `error`, value: msg }); }
+  expect( link, msg ).to.exist;
+
   await scope.afterStep(scope);
 });
 
-Then('I should see the link {string}', async (linkText) => {  // √
-  let [link] = await scope.page.$x(`//a[contains(text(), "${linkText}")]`);
-  expect(link).to.exist;
+Then('I should see the link to {string}', async ( url ) => {  // √
```

Comment:
The difference between "see the link xyz" and "see the link to xyz" being loosely match vs exact match isn't immediately obvious to me. Is there a way to reword this? Alternatively, you could deprecate the above function and keep just "see the link xyz". Or, you could merge the two functions somehow (first check for exact match, then if that doesn't exist, for loose match).

Reply:
It's actually even weirder than that and I'm planning to get rid of the previous one. That one uses the visible text of the link and is not language agnostic. This one uses the href of the link and _is_ language agnostic. The concern I currently have is over links with dynamic values. It's the same concern as the one about the download links and how to handle those (#340)